source/agora/node/Registry.d

```diff
@@ -216,10 +241,17 @@ public class NameRegistry: NameRegistryAPI
         TYPE payload_type = this.ensureValidPayload(data,
             this.zones[ZoneIndex.Validator].get(data.public_key));
```

Comment:
Just to note, `getValidatorInternal` invokes this even it knows the `.get` is empty

Reply:
True, that code definitely will need to be optimized in the future.